pkg/openid4ci/openid4ci.go

```diff
@@ -83,21 +95,18 @@ func NewInteraction(initiateIssuanceURI string, config *ClientConfig) (*Interact
 
 	setDefaults(config)
 
-	credentialOffer, err := getCredentialOffer(initiateIssuanceURI, config.HTTPClient, config.MetricsLogger)
+	httpClient := createHTTPClient(config.HTTPClientOpts)
```

Comment:
Note that I couldn't use the HTTP client as an interface anymore due to the OAuth2 library not accepting an interface, but rather the http.Client type. You'll see in this file I had to rework how we use the various HTTP client options.

Reply:
@vkubiv I've updated the go API OpenID4CI type to use an http.Client instead of passing in the individual parameters, and I moved the logic tor HTTP client creation into the wallet-sdk-gomobile layer (replacing the existing HTTP client wrapper type)